libflux/flux-core/src/semantic/convert.rs

```diff
@@ -2668,13 +2688,13 @@ mod tests {
 
     #[test]
     fn test_convert_monotype_record() {
-        let monotype = Parser::new("{ A with B: int }").parse_monotype();
+        let monotype = Parser::new("{ A with b: int }").parse_monotype();
```

Comment:
Yes, it should be fine due to that control. We may still want to consider changing what we consider a type variable though since it isn't very obvious that `A`,`B` etc are type variables. (And changing it should be fine since we control every specified type)